geo-data-server/src/main/java/org/gridsuite/geodata/server/GeoDataService.java

```diff
@@ -236,6 +236,36 @@ void saveLines(List<LineGeoData> linesGeoData) {
         lineRepository.saveAll(linesEntities);
     }
 
+    private LineGeoData getLineGeoDataWithEndSubstations(Map<String, LineGeoData> linesGeoDataDb, Map<String, SubstationGeoData> substationGeoDataDb, Line line) {
+        LineGeoData geoData = linesGeoDataDb.get(line.getId());
+        Substation sub1 = line.getTerminal1().getVoltageLevel().getSubstation();
+        Substation sub2 = line.getTerminal2().getVoltageLevel().getSubstation();
+        Coordinate substation1Coordinate = substationGeoDataDb.get(sub1.getId()).getCoordinate();
+        Coordinate substation2Coordinate = substationGeoDataDb.get(sub2.getId()).getCoordinate();
+
+        if (geoData == null || geoData.getCoordinates().isEmpty()) {
+            return new LineGeoData(line.getId(), sub1.getNullableCountry(), sub2.getNullableCountry(), "",
+                List.of(substation1Coordinate, substation2Coordinate));
+        }
+
+        boolean reverse = sub2.getId().equals(geoData.getSubstationStart());
+        return new LineGeoData(line.getId(), sub1.getNullableCountry(), sub2.getNullableCountry(),
+            geoData.getSubstationStart(),
+            addCoordinates(substation1Coordinate, geoData.getCoordinates(), substation2Coordinate, reverse));
+    }
+
+    private List<Coordinate> addCoordinates(Coordinate prepend, List<Coordinate> list, Coordinate append, boolean reverse) {
+        List<Coordinate> res = new ArrayList<>(list.size() + 2);
+        res.add(reverse ? prepend : append);
```

Comment:
I don't think this works.. This gives
when reversed:
append .. reverse(list) .. prepend
when not reversed:
append .. list .. prepend

but obviously the same substation can not be sometimes connected to the first pylon, sometimes to the last pylon

are there unit tests for this ?